python/services/dataservice/dmod/dataservice/dataset.py

```diff
@@ -0,0 +1,138 @@
+import re
+from pydantic import BaseModel, Field, UUID4, validator
+from datetime import datetime
+from uuid import uuid4
+from typing import Annotated, ClassVar, Optional, List, Union
+from enum import Enum
+
+from .models import DataCategory, DataDomain
+
+
+class DatasetType(Enum):
+    UNKNOWN = "UNKNOWN"
+    OBJECT_STORE = "OBJECT_STORE"
+    FILESYSTEM = "FILESYSTEM"
+
+
+class Dataset(BaseModel):
+    """
+    Rrepresentation of the descriptive metadata for a grouped collection of data.
+    """
+
+    _SERIAL_DATETIME_STR_FORMAT: ClassVar = "%Y-%m-%d %H:%M:%S"
+    _NAME_RE: ClassVar = re.compile(
+        "(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$"
+    )
+
+    name: Annotated[
```

Comment:
`typing.Annotated` was added in 3.9. Did we make the jump? I thought we were stuck in 3.8.

Reply:
Nice catch! I'll bring it in from `typing_extensions`... and switch my dev environment over to 3.8.

Reply:
bummer, right? I saw `Annotated` and got excited about this new feature I didn't know about and... then found out I couldn't use it.

I think there's some other goodies in `typing` that are worth using that we can't.

Reply:
`Self` is super nice